blog/2023-03-20-handler-slack-commands/index.md

```diff
@@ -0,0 +1,230 @@
+tags: [tutorial, slack, slackbot, commands, workflow]
+image: ./0-header.png
+---
+
+Slackbots are great tools to save time and reduce the need for manual input. They are also useful if you want operational teams to interact with custom-made workflows.
+
+<!--truncate-->
+
+Building a Slack commands handler requires a little bit of Windmill mastering, but it's also a perfect way to get familiar with Windmill, as some key features are used (scripts, worfklows, branches, approval steps).
+
+Also, know that non-technical users can totally build such tools. I myself am not a developer.
+
+:::info
+
+Windmill was not made for building slackbots. They are one of the side effects of our modular product.
+
+:::
+
+![Windmill Slack](./0-header.png "Windmill & Slack")
+
+:::tip After this article
+
+You will be able to build a Slackbot from which you can create events on your calendar, send emails and even generate them with Open-AI, all of this from Slack.
+
+:::
+
+<!--If self-hosted-->
+<!--If self-hosted/-->
+
+<br/>
+
+We've seen in a [previous article](2023-02-13-slack-integration/index.md) how to connect Slack with Windmill and have briefly explored how to build a first slackbot for a single command (using a script).
+
+Long story short, the first article explained how to connect Slack with Windmill and to create a `/windmill` command on Slack linked to a script hosted on Windmill. Handling several commands on your Slackbot uses just the same logic, but we'll use one single Windmill workflow instead of a script.
+
+[Workflows](https://www.windmill.dev/docs/getting_started/flows_quickstart) on Windmill are based on scripts. Workflows are the architecture that allows you to chain scripts with branches, loops etc. while connecting scripts' inputs to previous ouputs.
+
+![Windmill Workflows](./1-windmill-workflows.png "Windmill workflows")
+
+As we will use only one master workflow to handle all Slack commands, branches will be key: one branch for one Slack command. Let's get started.
+
+## Connecting a first workflow handler to your workspace
+
+Just like the [slack integration tutorial](2023-02-13-slack-integration/index.md), it all starts from your workspace settings. Once you connected Slack to your Windmill account, you will be offered to connect a script or flow to the `/windmill` Slack command. This time, pick "Create a flow to handle Slack commands".
+
+![Create Flow Handler](./2-create-flow-handler.png "Create a flow to handle Slack Commands")
+
+You will get to [this](https://hub.windmill.dev/flows/28/example-handler-for-slack-bot-commands) workflow to handle Slack commands. When linked to the `/windmill` command from the Windmill workspace settings, it will react to it on Slack.
+
+Before going deeper, let's take a look at this first workflow because its logic will be of use.
+
+![First Handler Workflow](./8-example-first-handler.png "By default commands handler workflow")
+
+### Inputs
+
+There are two workflow inputs:
```

Comment:
There are two workflow inputs that are passed to your flow everytime a command is emitted after you have setup the slack command integration in windmill.